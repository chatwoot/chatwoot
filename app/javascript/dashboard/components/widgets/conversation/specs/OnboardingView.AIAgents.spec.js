import { describe, it, expect, vi } from 'vitest';
import { mount } from '@vue/test-utils';
import AddAIAgentView from '../OnboardingView.AddAIAgent.vue';

vi.mock('../../../../routes/dashboard/settings/agentBots/Index.vue', () => ({
  default: {
    name: 'Index',
    template: '<div class="mock-index">Index Component</div>',
  },
}));

describe('OnboardingView.AddAIAgent.vue', () => {
  it('should render the component', () => {
    const wrapper = mount(AddAIAgentView);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('render the index when current step is equal to stepNumber', () => {
    const wrapper = mount(AddAIAgentView, {
      props: {
        currentStep: 1,
        stepNumber: 1,
      },
    });
    expect(wrapper.find('.mock-index').exists()).toBe(true);
  });

  it('should not render when the step is not the current step', () => {
    const wrapper = mount(AddAIAgentView, {
      props: {
        currentStep: 1,
        stepNumber: 2,
      },
    });
    expect(wrapper.find('.mock-index').exists()).toBe(false);
  });
});
