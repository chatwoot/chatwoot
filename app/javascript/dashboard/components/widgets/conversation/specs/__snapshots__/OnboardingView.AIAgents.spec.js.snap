// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OnboardingView.AddAIAgent.vue > should render the component 1`] = `
"<div>
  <div class="mock-index">Index Component</div>
</div>"
`;
